source/fuzz/fact_manager/irrelevant_value_facts.cpp

```diff
@@ -56,17 +58,79 @@ void IrrelevantValueFacts::AddFact(
   irrelevant_ids_.insert(fact.result_id());
 }
 
-bool IrrelevantValueFacts::PointeeValueIsIrrelevant(uint32_t pointer_id) const {
-  return pointers_to_irrelevant_pointees_ids_.count(pointer_id) != 0;
+bool IrrelevantValueFacts::PointeeValueIsIrrelevant(
```

Comment:
I'm not sure if we can consider pointees irrelevant even if the corresponding instructions are in dead blocks. Consider:
```
%1 = OpFunction %void None %type
%2 = OpLabel
%3 = OpVariable ...
...
     OpBranchConditional %true %4 %dead
%4 = OpLabel
...
%dead = OpLabel
%5 = OpCopyObject %pointer %3 ; can't be marked with PointeeValueIsIrrelevant
...
```

This is related to the fact that we can't determine whether two pointers point to the same memory location right now (#3723).

Reply:
You are quite right, I have now fixed it so that they are not considered irrelevant